pkg/plugin/GlobalPluginService.go

```diff
@@ -172,6 +167,23 @@ func (impl *GlobalPluginServiceImpl) GetAllGlobalVariables() ([]*GlobalVariable,
 			Type:        "cd",
 		},
 	}
+	var globalVariable *GlobalVariable
+	if appType == helper.Job {
+		globalVariable = &GlobalVariable{
+			Name:        "JOB_NAME",
+			Format:      string(repository.PLUGIN_VARIABLE_FORMAT_TYPE_STRING),
+			Description: "Name of the Job this pipeline resides in.",
+			Type:        "ci",
+		}
+	} else {
```

Comment:
make this APP_NAME as default and then assign job name if app type is job

Reply:
resolved